cmake/BundledLibArchive.cmake

```diff
@@ -0,0 +1,62 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+function(use_bundled_libarchive SOURCE_DIR BINARY_DIR)
+    # Define patch step
+    set(PC "${Patch_EXECUTABLE}" -p1 -i "${SOURCE_DIR}/thirdparty/libarchive/libarchive.patch")
+
+    # Define byproducts
+    if (WIN32)
```

Comment:
Just nitpicking, wonder if we could set the platform-specific suffix (and possibly prefix as well) somewhere at root level and just append that in all the cmake files to avoid these copy-pastes. (Perfectly fine as a follow-up)

Reply:
Not a very good idea in my opinion. Platform-specific suffixes are very package-dependent. While static libraries will always be .lib on Windows, the suffix actually contains other things (like the release version) in some packages, and may be different on other architecture (x86/x64). The process for figuring this out when creating a new third party dependency is building it manually and seeing for yourself what artifacts it generates (or looking at the build commands/files of the third party).
I am afraid if we abstract this system-wide, it will become a copy-pasted voodoo-programming feat, that will be used without actually figuring out what artifacts the third party generates on multiple OS-es/multiple architectures.

Reply:
Okay, I can live with this.